lib/src/controllers/payment_method_selector_controller.dart

```diff
@@ -70,7 +110,39 @@ class PaymentMethodSelectorController
       } else {
         error = e.toString();
       }
-      _setValue(value.copyWith(status: Status.error, errorMessage: error));
+      _setValue(value.copyWith(
+          capabilityLoadingStatus: Status.error,
+          capabilityErrorMessage: error));
+    }
+  }
+
+  /// Creates a source based on the collected data from the user.
+  Future<void> createSource() async {
+    try {
+      // Set the status to loading while creating the token
+      _setValue(value.copyWith(sourceLoadingStatus: Status.loading));
+
+      // Create the token using Omise API
+      final source = await omiseApiService.createSource(CreateSourceRequest(
+          amount: value.amount!,
+          currency: value.currency!,
+          type: value.selectedPaymentMethod!));
+
+      _setValue(value.copyWith(
+        source: source,
+        sourceLoadingStatus: Status.success,
+      ));
+    } catch (e) {
+      // Handle errors and update the state with an error message
+      var error = "";
+      log(e.toString());
```

Comment:
Can you help me understand how will this log behave if `e` is type of OmiseApiException?

Reply:
when `e` is of type `OmiseApiException` when can get the error message from the api using the `message` param, otherwise use whatever error is thrown.